### PR TITLE
updated keras backend with missing methods

### PR DESCRIPTION
**Pull request by al3xsh**
Hi,

The keras backend in the tensorflow repository is currently missing:

conv3d_transpose 
cumprod
cumsum
depthwise_conv2d
identity
local_conv1d
local_conv2d
logsumexp
tile

from the __init__.py file.

They are present in the _impl.backend package but haven't been brought through via imports in the __int__.py file.

Regards,

Alex

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

**Comment by yifeif** [comment]
Looks like you will also need to update the golden API. Could you run
    $ bazel build tensorflow/tools/api/tests:api_compatibility_test
    $ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test \
          --update_goldens True

**Comment by yifeif** [comment]
@martinwicke could you approve the API change? Thanks!

**Comment by martinwicke** [comment]
fine for API, assuming tests are fixed.

**Comment by drpngx** [comment]
@al3xsh mind updating the goldens as per @yifeif comment?

**Comment by al3xsh** [comment]
Apologies for the delay - I can't do this until I'm back in my office - which wont be until next week!

I'll try and do it as soon as possible after then!

Regards,

Alex

**Comment by al3xsh** [comment]
Massive apologies - I have been snowed under at work.  I aim to get this done today (or the weekend at the latest!).

Sorry!

Alex

**Comment by al3xsh** [comment]
@yifeif @drpngx @rmlarsen I have updated the goldens as per the request using the commands provided.  I'm not sure what to do with it now though!  Git tells me there is nothing to commit.

The output of "$ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True" is:

```
alex@ubuntu:~/tensorflow$ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True
s.s.
----------------------------------------------------------------------
Ran 4 tests in 0.000s

OK (skipped=2)
```

I'm sorry - I'm not familiar with the TensorFlow build process, so may well be messing something up.

Any ideas what to do with it now?

Regards,

Alex

**Comment by martinwicke** [comment]
Could you run those with Py2? The compatibility tests only do their thing with Py2 for technical reasons.

**Comment by al3xsh** [comment]
@martinwicke 

Thanks for the help.  I've installed python 2.7 in an anaconda environment, exported PYTHON_BIN_PATH to point to the python in the environment and rerun everything, but I still don't see any changes in git.  See log below:

```
(tf_dev_py2) alex@ubuntu:~/tensorflow$ export PYTHON_BIN_PATH="/home/alex/anaconda3/envs/tf_dev_py2/bin/python"
(tf_dev_py2) alex@ubuntu:~/tensorflow$ bazel build tensorflow/tools/api/tests:api_compatibility_test --incompatible_load_argument_is_label=false --local_resources 4096,.5,1.0
WARNING: /home/alex/tensorflow/tensorflow/core/BUILD:1795:1: in includes attribute of cc_library rule //tensorflow/core:framework_headers_lib: '../../external/nsync/public' resolves to 'external/nsync/public' not below the relative path of its package 'tensorflow/core'. This will be an error in the future. Since this rule was created by the macro 'cc_header_only_library', the error might have been caused by the macro implementation in /home/alex/tensorflow/tensorflow/tensorflow.bzl:1086:30
WARNING: /home/alex/tensorflow/tensorflow/contrib/learn/BUILD:15:1: in py_library rule //tensorflow/contrib/learn:learn: target '//tensorflow/contrib/learn:learn' depends on deprecated target '//tensorflow/contrib/session_bundle:exporter': No longer supported. Switch to SavedModel immediately.
WARNING: /home/alex/tensorflow/tensorflow/contrib/learn/BUILD:15:1: in py_library rule //tensorflow/contrib/learn:learn: target '//tensorflow/contrib/learn:learn' depends on deprecated target '//tensorflow/contrib/session_bundle:gc': No longer supported. Switch to SavedModel immediately.
INFO: Analysed target //tensorflow/tools/api/tests:api_compatibility_test (0 packages loaded).
INFO: Found 1 target...
Target //tensorflow/tools/api/tests:api_compatibility_test up-to-date:
  bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
INFO: Elapsed time: 0.528s, Critical Path: 0.00s
INFO: Build completed successfully, 1 total action
(tf_dev_py2) alex@ubuntu:~/tensorflow$ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True
s.s.
----------------------------------------------------------------------
Ran 4 tests in 0.000s

OK (skipped=2)
(tf_dev_py2) alex@ubuntu:~/tensorflow$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
(tf_dev_py2) alex@ubuntu:~/tensorflow$ 
```
Any ideas?

Regards,

Alex

**Comment by drpngx** [comment]
@al3xsh it's skipping the tests, so it looks like you're still running python3. Can you try `python2 blaze-bin/...`?

**Comment by al3xsh** [comment]
Hi @drpngx, I've tried that and it still says the same:

```
(tf_dev_py2) alex@ubuntu:~/tensorflow$ python --version
Python 2.7.14 :: Anaconda, Inc.
(tf_dev_py2) alex@ubuntu:~/tensorflow$ python2 bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True
s.s.
----------------------------------------------------------------------
Ran 4 tests in 0.001s

OK (skipped=2)
(tf_dev_py2) alex@ubuntu:~/tensorflow$ 
```

Sorry for being a pain - any more ideas? 

Alex

**Comment by drpngx** [comment]
@gunan any reason why the test is not running?
 Also, maybe we could make the test fail if `--update_goldens` is set to `True`.

**Comment by gunan** [comment]
I think you probably need to rerun configure to change the python version you are using. once configure is run, the environment variables are hardcoded and will override what you set unless you run configure.

The test will be skipped when run on python3. Instead of using `unittest.skip` we can just modify the first few lines of the test to fail in different scenarios.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
Not sure why this is running python3 still. 
It looks like we have `PY2AND3` that we could change to `PY2ONLY` for that rule?

**Comment by martinwicke** [comment]
PY2_AND_3 just tells bazel not to try and auto-convert the code to another python version in case of a mismatch (which never works anyway). @gunan we should definitely improve the output of this to make clear what is happening for Py3.

@al3xsh can you reconfigure for Py2 and run the update? Sorry for the complication.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jhseu: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by al3xsh** [comment]
Hi @drpngx and @martinwicke,

I have now managed to run the bazel build process as requested and have pushed the changes.  The results of the commands were:

```
(tf_dev) alex@ubuntu:~/tensorflow$ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True
WARNING:tensorflow:VARIABLES collection name is deprecated, please use GLOBAL_VARIABLES instead; VARIABLES will be removed after 2017-03-02.
ERROR:tensorflow:TensorFlow API backwards compatibility test
This test ensures all changes to the public API of TensorFlow are intended.

If this test fails, it means a change has been made to the public API. Backwards
incompatible changes are not allowed. You can run the test as follows to update
test goldens and package them with your change.

    $ bazel build tensorflow/tools/api/tests:api_compatibility_test
    $ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test \
          --update_goldens True

You will need an API approval to make changes to the public TensorFlow API. This
includes additions to the API.

ERROR:tensorflow:1 differences found between API and golden.
ERROR:tensorflow:Issue 1	: Change detected in python object: tensorflow.keras.backend.
WARNING:tensorflow:Golden file update requested!
All test failures have been skipped, see the logs for detected diffs.
This test is now going to write new golden files.
Make sure to package the updates together with your change.

You will need an explicit API approval. This may take longer than a normal
review.

..WARNING:tensorflow:Unexpected hidden op name: ConjugateTranspose
WARNING:tensorflow:Unexpected hidden op name: ScaleImageGrad
WARNING:tensorflow:Unexpected hidden op name: MutexAcquire
WARNING:tensorflow:Unexpected hidden op name: TensorArrayPackV2
WARNING:tensorflow:Unexpected hidden op name: AccumulateNV2
WARNING:tensorflow:Unexpected hidden op name: ReaderWorkQueueLengthV2
WARNING:tensorflow:Unexpected hidden op name: SparseSelectLastK
WARNING:tensorflow:Unexpected hidden op name: Mutex
WARNING:tensorflow:Unexpected hidden op name: TensorArrayPackV3
WARNING:tensorflow:Unexpected hidden op name: AddV2
WARNING:tensorflow:Unexpected hidden op name: HistogramFixedWidth
WARNING:tensorflow:Unexpected hidden op name: TensorArrayUnpackV3
WARNING:tensorflow:Unexpected hidden op name: ReaderWorkQueueLength
WARNING:tensorflow:Unexpected hidden op name: TensorArrayUnpackV2
WARNING:tensorflow:Unexpected hidden op name: MutexRelease
..
----------------------------------------------------------------------
Ran 4 tests in 4.517s

OK
(tf_dev) alex@ubuntu:~/tensorflow$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   tensorflow/tools/api/golden/tensorflow.keras.backend.pbtxt
	modified:   tensorflow/workspace.bzl

no changes added to commit (use "git add" and/or "git commit -a")
(tf_dev) alex@ubuntu:~/tensorflow$ git add .
(tf_dev) alex@ubuntu:~/tensorflow$ git commit -m "updated goldens as per discussion"
```

Does this look ok?

Regards,

Alex

**Comment by martinwicke** [comment]
I the process, I fear you have replicated all commits from master during some period. Can you clean the commit history?

**Comment by al3xsh** [comment]
@martinwicke 

I'm sorry - I have no idea what I have done :( 

I had managed to mess up the name in one of the git commits, which meant that the CLA got messed up.  In trying to fix that name I appear to have royally screwed up the commit history!  

Would it be easier to just start again and submit a new PR?

Sorry for being a pain :(

Alex

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jhseu: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jhseu** [comment]
Yeah, mind submitting a new PR? We'll need to address both the CLA and the merge conflicts before we can merge. Thanks!
